tests/unit/stateService.js

```diff
@@ -222,6 +226,239 @@ tap.test('States DB Services', {autoend: true}, t => {
+          lookupNetworkError: false,
+          lookupHttpStatus: 301,
+          system: {
+            timestamps: {
+              endpointStart: endpointStart1,
+              endpointEnd: endpointEnd1,
+              endpointDuration: {
+                milliseconds: endpointDurationMs1
+              }
+            }
+          }
+        }
+
+        const endpointStart2 = DateTime.utc().toISO()
+        const endpointEnd2 = DateTime.utc().plus(50).toISO()
+        const endpointDurationMs2 = 50
+        const stateObject2 = {
+          _endpointReference: endpointId,
+          lookupNetworkError: true,
+          lookupHttpStatus: 500,
+          system: {
+            timestamps: {
+              endpointStart: endpointStart2,
+              endpointEnd: endpointEnd2,
+              endpointDuration: {
+                milliseconds: endpointDurationMs2
+              }
+            }
+          }
+        }
+
+        const endpointStart3 = DateTime.utc().minus({minutes: 60}).toISO()
+        const endpointEnd3 = DateTime.utc()
+          .minus({minutes: 59, milliseconds: 500})
+          .toISO()
+        const endpointDurationMs3 = 500
+        const stateObject3 = {
+          _endpointReference: endpointId,
+          lookupNetworkError: false,
+          lookupHttpStatus: 202,
+          system: {
+            timestamps: {
+              endpointStart: endpointStart3,
+              endpointEnd: endpointEnd3,
+              endpointDuration: {
+                milliseconds: endpointDurationMs3
+              }
+            }
+          }
+        }
+
+        //create first state record
+        await createEndpointState(stateObject1)
+          .then(data => {
+            t.ok(data.id)
+          })
+          .catch(error => {
+            t.fail(`Should not reach here. ${error.message}`)
+          })
+        // create latest state record
+        await createEndpointState(stateObject2)
+          .then(data => {
+            t.ok(data.id)
+          })
+          .catch(error => {
+            t.fail(`Should not reach here. ${error.message}`)
+          })
+        //create third state record
+        await createEndpointState(stateObject3)
+          .then(data => {
+            t.ok(data.id)
+          })
+          .catch(error => {
+            t.fail(`Should not reach here. ${error.message}`)
+          })
+
+        await readLatestEndpointStateById(endpointId, 'exclude', ['301'])
+          .then(data => {
+            t.ok(data._id)
+            t.deepEquals(data._endpointReference, endpointId)
+            t.equals(data.system.timestamps.endpointStart, endpointStart1)
+            t.equals(data.system.timestamps.endpointEnd, endpointEnd1)
+            t.equals(
+              data.system.timestamps.endpointDuration.milliseconds,
+              endpointDurationMs1
+            )
+            t.equals(data.lookupNetworkError, false)
+            t.equals(data.lookupHttpStatus, 301)
```

Comment:
@MattyJ007 not sure if im a lil confused, or if this is a bug, but the function fetching the results has a filter that should `exclude` the `301` status code. Should this result then not be the `500` status code state record?

Reply:
Exclude and 301 aren't related.
Network errors are completely separate from httpStatuses.

When we're filtering networkErrors thefilter options are: `include`, `exclude`, or `no-filter`

When filtering httpStatuses you have options such as: `200`, `2xx`, or `*`

Reply:
In the test, we're excluding network error states.
With httpStatuses we only want to return states with a status of 301

Reply:
aaah right... I think I read this in the wrong context. Was reading this as Exclude 301 statuses, but the second argument is for the networkErrors options (Include/exclude etc), and the 3rd argument is for the httpStatuses array to include